### DISCUSSION TITLE
Kamal pushes Docker images to public Docker Hub by default

### DISCUSSION THREAD

**Discussion by r3cha** [basecamp/kamal#1128] [General]
Hey everyone,

I just tried Kamal and noticed that it automatically builds and pushes Docker images to the public Docker Hub registry with default configuration. I didn’t see any warning about this in the docs, and I’m thinking some users might end up with their images publicly available without realizing it.

It’d be great to add a quick head-up in the documentation, suggesting users set up a private registry (or make their Docker Hub repo private) before deploying. That way, people don’t accidentally expose their images.

Thanks for the awesome tool!

**Comment by viktorianer** [comment]
I’m also trying to figure out if and how it’s possible to set up a private repository, but I haven’t found a documentation for that. Need to read code now 💯.

**Reply by kjlape** [reply]
Take a look at this: https://gist.github.com/pingortle/040098454af630cc6cf6fa78bed9b018

It's not perfect, but it gets around the chicken-or-the-egg problem of bootstrapping a private registry using kamal without using a cloud provider.

**Comment by nickhammond** [comment]
Kamal just pushes your image. It doesn't care if it's a public or private image, this is on the Docker Hub side. 

You can change the default behavior by going into your repo settings https://hub.docker.com/repository-settings/default-privacy or with each individual repository.
<img width="943" alt="image" src="https://github.com/user-attachments/assets/a045a7ba-2bad-4298-ab48-3c5a44a83ee0">

**Reply by r3cha** [reply]
@nickhammond understand that, free account allow have only one private repo and they are public opt-in at docker hub. I just think that default way should be more secure. Would be great to notice people about that in documentation

**Reply by nickhammond** [reply]
Gotcha, feel free to open a PR here https://github.com/basecamp/kamal-site/blob/main/docs/configuration/docker-registry.md that mentions it.

**Reply by r3cha** [reply]
@nickhammond done https://github.com/basecamp/kamal-site/pull/137 please add comments to text if you think if might improved
